apps/src/music/settings.ts

```diff
@@ -0,0 +1,66 @@
+import {useEffect, useState} from 'react';
+
+import {BLOCKLY_THEME, Themes} from '@cdo/apps/blockly/constants';
+import {commonI18n} from '@cdo/apps/types/locale';
+
+import {getBaseName, setWorkspaceTheme} from '../blockly/utils';
+import {Setting} from '../lab2/views/components/Settings/SettingsDropdown';
+import UserPreferences from '../lib/util/UserPreferences';
+
+const blockThemeOptions = [
+  {
+    value: Themes.MODERN,
+    text: commonI18n.blocklyModernTheme(),
+  },
+  {
+    value: Themes.HIGH_CONTRAST,
+    text: commonI18n.blocklyHighContrastTheme(),
+  },
+  {
+    value: Themes.PROTANOPIA,
+    text: commonI18n.blocklyProtanopiaTheme(),
+  },
+  {
+    value: Themes.DEUTERANOPIA,
+    text: commonI18n.blocklyDeuteranopiaTheme(),
+  },
+  {
+    value: Themes.TRITANOPIA,
+    text: commonI18n.blocklyTritanopiaTheme(),
+  },
+];
+
+export function useMusicSettings(): Setting[] {
+  const [selectedTheme, setSelectedTheme] = useState<string | null>(null);
+
+  useEffect(() => {
+    new UserPreferences()
+      .getBlocklyTheme(() =>
+        getBaseName(
+          (localStorage.getItem(BLOCKLY_THEME) || Themes.MODERN) as Themes
+        )
+      )
+      .then((theme: string) => {
+        setSelectedTheme(theme);
```

Comment:
I may have missed this in a previous PR, but we do need to do anything make sure the rendered workspace is using the theme loaded from user preferences?